pallets/subtensor/tests/math.rs

```diff
@@ -0,0 +1,2344 @@
+            I32F32::from_num(9),
+            I32F32::from_num(9),
+            I32F32::from_num(9),
+        ],
+        epsilon,
+    );
+    assert_vec_compare(
+        &matmul(&w, &[I32F32::from_num(-1.0); 3]),
+        &[
+            I32F32::from_num(-3),
+            I32F32::from_num(-3),
+            I32F32::from_num(-3),
+        ],
+        epsilon,
+    );
+    let w: Vec<Vec<I32F32>> = vec![vec![I32F32::from_num(-1.0); 3]; 3];
+    assert_vec_compare(
+        &matmul(&w, &[I32F32::from_num(1.0); 3]),
+        &[
+            I32F32::from_num(-3),
+            I32F32::from_num(-3),
+            I32F32::from_num(-3),
+        ],
+        epsilon,
+    );
+    assert_vec_compare(
+        &matmul(&w, &[I32F32::from_num(2.0); 3]),
+        &[
+            I32F32::from_num(-6),
+            I32F32::from_num(-6),
+            I32F32::from_num(-6),
+        ],
+        epsilon,
+    );
+    assert_vec_compare(
+        &matmul(&w, &[I32F32::from_num(3.0); 3]),
+        &[
+            I32F32::from_num(-9),
+            I32F32::from_num(-9),
+            I32F32::from_num(-9),
+        ],
+        epsilon,
+    );
+    assert_vec_compare(
+        &matmul(&w, &[I32F32::from_num(-1.0); 3]),
+        &[
+            I32F32::from_num(3),
+            I32F32::from_num(3),
+            I32F32::from_num(3),
+        ],
+        epsilon,
+    );
+    let w: Vec<Vec<I32F32>> = vec![
+        vec![I32F32::from_num(1.0); 3],
+        vec![I32F32::from_num(2.0); 3],
+        vec![I32F32::from_num(3.0); 3],
+    ];
+    assert_vec_compare(
+        &matmul(&w, &[I32F32::from_num(0.0); 3]),
+        &[
+            I32F32::from_num(0.0),
+            I32F32::from_num(0.0),
+            I32F32::from_num(0.0),
+        ],
+        epsilon,
+    );
+    assert_vec_compare(
+        &matmul(&w, &[I32F32::from_num(2.0); 3]),
+        &[
+            I32F32::from_num(12),
+            I32F32::from_num(12),
+            I32F32::from_num(12),
+        ],
+        epsilon,
+    );
+    let w: Vec<Vec<I32F32>> = vec![
+        vec![
+            I32F32::from_num(1),
+            I32F32::from_num(2),
+            I32F32::from_num(3)
+        ];
+        3
+    ];
+    assert_vec_compare(
+        &matmul(&w, &[I32F32::from_num(0.0); 3]),
+        &[
+            I32F32::from_num(0.0),
+            I32F32::from_num(0.0),
+            I32F32::from_num(0.0),
+        ],
+        epsilon,
+    );
+    assert_vec_compare(
+        &matmul(&w, &[I32F32::from_num(2.0); 3]),
+        &[
+            I32F32::from_num(6),
+            I32F32::from_num(12),
+            I32F32::from_num(18),
+        ],
+        epsilon,
+    );
+}
+
+#[test]
+fn test_math_fixed_to_u16() {
+    let expected = u16::MIN;
+    assert_eq!(fixed_to_u16(I32F32::from_num(expected)), expected);
+
+    let expected = u16::MAX / 2;
+    assert_eq!(fixed_to_u16(I32F32::from_num(expected)), expected);
+
+    let expected = u16::MAX;
+    assert_eq!(fixed_to_u16(I32F32::from_num(expected)), expected);
+}
```

Comment:
maybe test handling when we are over the `u16` capacity? does it saturate?

Reply:
noooooooot my testtt